### PR TITLE
fix(filtering): Scroll chips before remove #5304

### LINKED ISSUE #5304: Error is thrown when remove the last filter chip
## Description  

 * igniteui-angular version: 7.3.x ; 8.1.x
 * browser: all

## Steps to reproduce  

1. Open the [sample](https://stackblitz.com/edit/angular-pubv21)
2. Apply a few filtering expressions until the left and right arrows are shown
3. Go to the last added filter chip 
4. Click to remove the chip

## Result  
What is the actual result after following the steps to reproduce?
image

## Expected result  
What is the expected result after following the steps to reproduce?
No errors should be thrown and the chip should be removed.

### PR DESCRIPTION
**Pull request by astaev**
Closes #5304

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Comment by tachojelev** [comment]
There seems to be an additional behavior that, which presents itself in not refreshing the filter row entirely when removing a chip:

Steps to reproduce:
1. Set the grid's width of the Grid Column Groups dev demo to 800px.
2. Open the Grid Column Groups dev demo.
3. Click the Filter chip for the ContactTitle column.
4. Add four condition chips.
5. Scroll to the right-most chip and remove it with the X icon.

**Result:** Chips are not rearranged properly.
image

File: projects/igniteui-angular/src/lib/grids/filtering/grid-filtering-row.component.ts [hunk 1]
```diff
@@ -753,7 +753,8 @@ export class IgxGridFilteringRowComponent implements AfterViewInit {
         const containerRect = this.container.nativeElement.getBoundingClientRect();
 
         for (let index = 0; index < chipAraeChildren.length; index++) {
-            if (Math.ceil(chipAraeChildren[index].getBoundingClientRect().left) < Math.ceil(containerRect.left)) {
+            if (Math.ceil(chipAraeChildren[index].getBoundingClientRect().left) < Math.ceil(containerRect.left) &&
```
Comment (DiyanDimitrov) [inline projects/igniteui-angular/src/lib/grids/filtering/grid-filtering-row.component.ts]: The first condition seems unnecessary. If the second is true, the first one will always be true.

Comment (astaev) [reply projects/igniteui-angular/src/lib/grids/filtering/grid-filtering-row.component.ts]: Done

File: projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts [hunk 2]
```diff
@@ -3249,6 +3249,73 @@ describe('IgxGrid - Filtering Row UI actions', () => {
...
+            expect(chips.length).toBe(4);
+
+            const leftArrowButton = GridFunctions.getFilterRowLeftArrowButton(fix).nativeElement;
+            expect(leftArrowButton).toBeTruthy('Left scroll arrow should be visible');
+
+            const rightArrowButton = GridFunctions.getFilterRowRightArrowButton(fix).nativeElement;
+            expect(rightArrowButton).toBeTruthy('Right scroll arrow should be visible');
+            expect(grid.rowList.length).toBe(2);
+
+            try {
```
Comment (DiyanDimitrov) [inline projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts]: Try calling the `onChipRemoved` method directly and use `toThrow()`.

**Pull request merged**